internal/xds/server/kubejwt/tokenreview.go

```diff
@@ -56,19 +55,21 @@ func (i *JWTAuthInterceptor) validateKubeJWT(ctx context.Context, token string)
 		return fmt.Errorf("token is not authenticated")
 	}
 
-	// TODO: (cnvergence) define a better way to check if the token is coming from the correct node
-	if tokenReview.Status.User.Extra != nil {
-		podName := tokenReview.Status.User.Extra[authPodNameKey]
-		if podName[0] == "" {
-			return fmt.Errorf("pod name not found in token review response")
-		}
-		parts := strings.Split(podName[0], "-")
-		irKey := fmt.Sprintf("%s/%s", parts[1], parts[2])
+	// Check if the service account name in the JWT token exists in the cache to verify that the token
+	// is valid for an Envoy proxy managed by Envoy Gateway.
+	// example: "system:serviceaccount:default:envoy-default-eg-e41e7b31"
+	parts := strings.Split(tokenReview.Status.User.Username, ":")
+	if len(parts) != 4 {
+		return fmt.Errorf("invalid username format: %s", tokenReview.Status.User.Username)
+	}
+	sa := parts[3]
 
-		if !i.cache.SnapshotHasIrKey(irKey) {
-			return fmt.Errorf("pod %s not found in cache", podName)
+	irKeys := i.cache.GetIrKeys()
+	for _, irKey := range irKeys {
+		if proxy.ExpectedResourceHashedName(irKey) == sa {
```

Comment:
Nit: Could we add a local `irKeyToServiceAccountName` method in this file instead? I think this would be better because:
* It’s more straightforward and easier to read
* I'd prefer to avoid importing the infra package into the xds package if possible

Reply:
we can add it, I will add the comment where it came from